libmattermost.c

```diff
@@ -945,6 +945,56 @@ mm_fetch_url(MattermostAccount *ma, const gchar *url, const gchar *postdata, Mat
 	g_free(cookies);
 }
 
+static void
+mm_send_email_cb(PurpleBuddy *buddy)
+{
+	const gchar *email = purple_blist_node_get_string(PURPLE_BLIST_NODE(buddy),"email");
+	const gchar *first_name = purple_blist_node_get_string(PURPLE_BLIST_NODE(buddy),"first_name");
+	const gchar *last_name = purple_blist_node_get_string(PURPLE_BLIST_NODE(buddy),"last_name");
+	gchar *full_email = g_strdup("\"");
+	
+	if (first_name) {
+		full_email = g_strconcat(full_email,first_name," ",NULL);
+	}
+	if (last_name) {
+		full_email = g_strconcat(full_email,last_name," ",NULL);
+	}
+	full_email = g_strconcat(full_email,"<",email,">",NULL);
+	full_email = g_strconcat(full_email,"\"",NULL);
+
+	gchar *cmd_line = g_strdup_printf(
+#ifdef _WIN32
+		"cmd /c start"
+#else
+		"xdg-open"
+#endif
+		" mailto:%s", full_email);
+		
+	g_spawn_command_line_async(cmd_line, NULL);
+	g_free(full_email);
+	g_free(cmd_line);
+}
+
+static GList *
+mm_buddy_menu(PurpleBuddy *buddy)
+{
+	GList *menu = NULL;
+	if (purple_blist_node_get_string(PURPLE_BLIST_NODE(buddy),"email")) {
+		PurpleMenuAction *action = purple_menu_action_new(_("Email Buddy"),PURPLE_CALLBACK(mm_send_email_cb),NULL, NULL);
+		menu = g_list_append(menu, action);
+	}
+	return menu;
+}
+
+static GList *
+mm_blist_node_menu(PurpleBlistNode *node)
+{
+	if(PURPLE_BLIST_NODE_IS_BUDDY(node)) {
+		return mm_buddy_menu((PurpleBuddy *) node);

```

Comment:
Might need to use the cast function `PURPLE_BUDDY(node)` instead for better purple3 compatibility